### PR TITLE
cabana: Improve seeking and zooming

### PR DESCRIPTION
**Pull request by deanlee**
resolve  https://github.com/commaai/openpilot/discussions/26091#discussioncomment-9289543 :

1. Immediately update the timeline to the current seeked-to time after seeking.
2. Improve camera caching by storing decoded frames within a 2-second window. This significantly boosts performance, especially when zooming into short time ranges, as frames can be retrieved from the cache instead of being decoded with ffmpeg.
3. Immediately return after calling `seekTo()` in `ChartsWidget::updateState()` to prevent drawing incorrect timelines in `Chart::updatePlot()`.

The comparison after zooming in to 100 milliseconds (Real-time recording) :
**before** (Seems stuck because of camera blocking and incorrect updatePlot, though it's not truly hanging) :

[Kazam_screencast_00145.webm](https://github.com/commaai/openpilot/assets/27770/b8875999-e6f4-446c-938c-19af6af35fbd)

**after**:

[Kazam_screencast_00144.webm](https://github.com/commaai/openpilot/assets/27770/6381c3a6-2525-46df-8895-27e942d17c83)

### CODE REVIEW

File: tools/cabana/chart/chart.cc [hunk 1]
```diff
@@ -511,7 +512,7 @@ void ChartView::mouseReleaseEvent(QMouseEvent *event) {
     if (rubber->width() <= 0) {
       // no rubber dragged, seek to mouse position
       can->seekTo(min);
-    } else if (rubber->width() > 10 && (max - min) > 0.01) { // Minimum range is 10 milliseconds.
+    } else if (rubber->width() > 10 && (max - min) > MIN_ZOOM_SECONDS) {
```
Comment (sshane) [inline tools/cabana/chart/chart.cc:515]: I could see it being useful to zoom in to 20-50ms, can we leave it as is?

**Comment by sshane** [comment]
I also had hanging when zooming in with wide ranges (~minutes), which seems fixed with this!

**Comment by sshane** [comment]
I noticed I can't press the back mouse button nearly as quickly to zoom out any more. It would also be nice to redo zoom actions (though not necessarily in this PR!)

File: tools/cabana/chart/chart.cc (hunk 1 above)
Comment (deanlee) [reply tools/cabana/chart/chart.cc:515]: reverted to 10 ms

**Comment by deanlee** [comment]
> I noticed I can't press the back mouse button nearly as quickly to zoom out any more. It would also be nice to allow redo zoom actions with the forward mouse button (though not necessarily in this PR!)

Since I don't have a back button on my mouse right now, I can't check it. But from what I can see, there don't seem to be any issues that would cause the back button to stop working. I suggest adding support for the forward button and fixing any possible issues with the back button in the next PR.

**Pull request merged**